Make Machine and Value generic over the Memory (#33)
RalfJung/minirust: lang/syntax.md

}
```

> In order to define expressions, we first need to define constants.

Reviewer: ```suggestion
We also need to define constants (a strict subset of `Value`).
```

And finally, the syntax of expressions:

```rust

> /// Constants are Values, but are not allowed to be Value::Ptr | Value::Union.

Reviewer: The comment is not accurate -- they can still be a pointer with `None` provenance.

Reviewer: ```suggestion
/// Constants are Values, but cannot have provenance.
/// (We actually entirely disallow Value::Ptr | Value::Union in the well-formedness checker.)
```
